eventuals/collect.h

```diff
@@ -5,50 +5,62 @@
 #include "eventuals/loop.h"
 #include "eventuals/type-traits.h"
 
-// NOTE: The following include is a rather large one.
-#include "google/protobuf/repeated_field.h"
-
 ////////////////////////////////////////////////////////////////////////
 
 namespace eventuals {
 
 ////////////////////////////////////////////////////////////////////////
 
```

Comment:
I like this approach!

What if we provided an extension point called `EventualsCollect` instead that let us not have to provide the same "insert function" for all non-STL types.

For example, we could have (very much pseudo code, I have not compiled this):

```cpp
template <typename T, typename U, std::enable_if_t<std::is_convertible_v<U, T>>>
void EventualsCollect(google::protobuf::RepeatedField<T>& repeated, U&& u) {
  repeated.Add(std::forward<U>(u));
}
```

Then when we get a value we want to insert we can just do:

`EventualsCollect(data, std::forward<decltype(value)>(value));`

Then we can implement an `EventualsCollect` for STL things like so:

```cpp
template <typename Collection, typename T, std::enable_if_t<HasEmplaceBack<Collection>::value>>
void EventualsCollect(Collection& collection, T&& t) {
  collection.emplace_back(std::forward<T>(t));
}

template <typename Collection, typename T, std::enable_if_t<HasInsert<Collection>::value>>
void EventualsCollect(Collection& collection, T&& t) {
  collection.insert(std::forward<T>(t));
}
```